### ISSUE TITLE
[Docs] Features examples

### ISSUE THREAD

**Issue by wwalc** [ckeditor/ckeditor5#514]
As agreed, we should showcase features in a separate section. While there is a chance we will be progressively adding samples for all features, there already some features which should be explained sooner than later, because of being hard to discover or sometimes confusing:

- [x] Autoformatting
- [x] Images (with custom styling)
- [x] Headings (with an example of adding support for H1) -?
- [x] Keyboard support (done: 904d832)
- [x] Markdown
- [x] Read-only support -?
- [ ] Content filtering -?
- [x] Localizability (UI Language) -?

When creating the examples we could/should not only prepare a demo where the feature can be tested, but also provide all the information for the user allowing him to understand all the functionality it brings. It may happen that the description will turn out to be long (e.g. the list of supported keyboard shortcuts). Thus, we may end up with a complementary [guide with functionality overview](http://docs.ckeditor.com/#!/guide/dev_features) like in CKEditor 4.

Features which could deserve some additional documentation:
- [x] Autoformatting - list all supported formats.
- [x] Keyboard shortcuts - list all keyboard shortcuts.
- [x] Headings and heading levels philosophy.
- [x] Styling Images.
- [ ] Uploading images.
- [ ] Responsive images.
- [ ] Content filtering - the simplest possible introduction on how to alter schema ("configure CKEditor Build") to achieve functionality similar to `extraAllowedContent` / `allowedContent: true`.
- [x] Localizability, how to change language.

**Comment by Reinmar** [comment]
We need a sample content for examples.

**Comment by Reinmar** [comment]
Please check the first feature's guide: https://github.com/ckeditor/ckeditor5-autoformat/commit/ec87d0f1d3d86a3794c7b001fff7c2181cdc6e04.

**Comment by scofalik** [comment]
I don't know how large you planned those guides to be, but maybe it's a good idea to give a bit more insight about creating own formatters?

**Comment by Reinmar** [comment]
Definitely. But it's not a priority now. We don't have enough time and the API will change so I avoided going into any details.

**Comment by Reinmar** [comment]
And another one: https://github.com/ckeditor/ckeditor5-heading/commit/37be919e31056744716703cd2fd9f4f8f039287e

**Comment by Reinmar** [comment]
I've just realised that to create samples in the image guide I need to figure out how to load the stylesheet just once. This will require some tricks with the snippet adapter, but should be possible.

**Comment by Reinmar** [comment]
Uploading images and responsive images are blocked by unclear future of the upload feature.

**Comment by Reinmar** [comment]
I added read-only and UI language guides. I also need to update customising builds guide.

**Comment by AnnaTomanek** [comment]
Question: what do we really want to have in the Features section?

1. A sample with description/documentation.
2. Just a sample with minimal documentation (and then documentation goes where? And what is the Examples section for?)
3. Just documentation (while the sample lands in Examples).

At the moment it's inconsistent so I suggest we make it clear before we create too big a mess for us to deal with later.

**Comment by Reinmar** [comment]
In the Features section we wanted to document what each feature is, how it can be configured, etc. The examples there are showcasing feature options or how a feature works in general.

In the Examples section we want to have general examples of the builds that we'll have and some integrations (copied from the ckeditor.com page).

These are different scenarios. In both cases you may have live editors, but in Examples we'll be focused on "The editor" and in Features on each feature.

E.g. it'd be hard to showcase all image feature options in general Examples. The same with all basic styles (out of which only bold and italic are usually loaded). Hence, Features section where all packages and other distinctive platform features (such as keystrokes or content filtering) can be explained in details.

This will allow keeping Examples section as simple as possible, focused on the live demos included there instead of textual content.

**Comment by Reinmar** [comment]
A basic guide for the Markdown DP: https://github.com/ckeditor/ckeditor5-markdown-gfm/blob/master/docs/features/markdown.md.

**Comment by Reinmar** [comment]
Let's say this is done. And what's not done was extracted to https://github.com/ckeditor/ckeditor5/issues/584.
